doc/source/f2py/code/CMakeLists.txt

```diff
@@ -0,0 +1,65 @@
+### setup project ###
+cmake_minimum_required(VERSION 3.21.4)
+
+project(fibby
+  VERSION 1.0
+  DESCRIPTION "FIB module"
+  LANGUAGES C Fortran
+  )
+
+# Safety net
+if(PROJECT_SOURCE_DIR STREQUAL PROJECT_BINARY_DIR)
+  message(
+    FATAL_ERROR
+      "In-source builds not allowed. Please make a new directory (called a build directory) and run CMake from there.\n"
+  )
+endif()
+
+# Grab Python, 3.12 onwards
+find_package (Python3 COMPONENTS Interpreter Development NumPy)
+
+# Grab the variables from a local Python installation
+# NumPy headers
+execute_process(
+  COMMAND "${PYTHON_EXECUTABLE}"
+  -c "import numpy; print(numpy.get_include())"
+  OUTPUT_VARIABLE NP_INCLUDE_DIR
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+)
```

Comment:
note that this is also available in the `Python_NumPy_INCLUDE_DIRS` variable.

Reply:
Excellent points, modified accordingly